### PR TITLE
diagnose port-conflict command: remove android code and add case where ddagent user cannot read proc info

### PR DESCRIPTION
**Pull request by mwdd146980**
### What does this PR do?

This PR 1) removes android code that was used when we pulled this code from tailscale/tailscale and 2) adds a condition in ports.go that checks if the port process ID is read as 0. 

### Motivation

During QA we found that on a Debian VM, the process name and PID are only returned if the process that is binding a used port is launched by the same user as the user that launched the Agent.  

### Describe how to test/QA your changes

Check that the diagnose suite works properly for Linux and Darwin:
- when the agent isn't running and there is no port issue
- when the agent is running and there is no port issue (the agent is detected as being the agent, so no issue is raised)
- when some port is used by another process, it is reported
- You can use netcat to bind to a port:
`netcat -nvlp 6062`

@misteriaud found that the process name and PID are only given if the process that binds the port is launched by the same user as the one who launched the Agent:

run `netcat -nvlp 6062` (admin user) -> `Diagnosis: Required port 6062 is already used by '' process (PID=0) for tcp.`
run `sudo netcat -nvlp 6062` (root user) -> `Diagnosis: Required port 6062 is already used by '' process (PID=0) for tcp.`
run `sudo -u dd-agent netcat -nvlp 6062` (dd-agent user) -> `Diagnosis: Required port 6062 is already used by 'netcat' process (PID=183666) for tcp.`

This PR should modify this behavior so that the message is more generic and does not display a blank "" with PID=0. Expected output is:

run `netcat -nvlp 6062` (admin user) -> `Diagnosis: Required port 6062 is already used by an undefined process. The Agent user may not have permissions to read additional process info.`
run `sudo netcat -nvlp 6062` (root user) -> `Diagnosis: Required port 6062 is already used by an undefined process. The Agent user may not have permissions to read additional process info.`
run `sudo -u dd-agent netcat -nvlp 6062` (dd-agent user) -> `Diagnosis: Required port 6062 is already used by 'netcat' process (PID=183666) for tcp.`

### CODE REVIEW

File: pkg/diagnose/ports/ports.go [hunk 1]
```diff
@@ -72,6 +72,15 @@ func DiagnosePortSuite() []diagnosis.Diagnosis {
 			continue
 		}
 
+		if port.Pid == 0 {
```
Comment (dustmop) [inline pkg/diagnose/ports/ports.go:75]: Could you add a comment here describing the OS's behavior regarding the user running the command? Also, is this specific to Debian, and if so please comment about that as well.

**Review by dustmop — APPROVED** [review]
Is this behavior specific to Debian, or was that just where it was discovered? Adding a comment about this particular case would be helpful. Once that's completed, looks good to me!

File: pkg/diagnose/ports/ports.go (hunk 1 above)
Comment (mwdd146980) [reply pkg/diagnose/ports/ports.go:75]: It was discovered on Debian, but I also replicated the behavior on Ubuntu and RHEL, so I think it's pretty generic. I'll include that info in my comment.

**Pull request merged**